plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -18,6 +18,7 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
```

Comment:
I have managed to understand now the problem you're pointing towards by using multiple delete files.

Now it is easier  in debug what the problem is:


```
    @Test
    public void testMultipleEqualityDeletes()
            throws Exception
    {
        String tableName = "test_equality_delets_different_schemas_" + randomNameSuffix();
        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
        Table icebergTable = updateTableToV2(tableName);
        Assertions.assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
        Path metadataDir = new Path(metastoreDir.toURI());
        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(SESSION);


        List<String> firstDeleteFileColumns = ImmutableList.of("regionkey");
        Schema deleteRowSchema = icebergTable.schema().select(firstDeleteFileColumns);
        List<Integer> equalityFieldIds = firstDeleteFileColumns.stream()
                .map(name -> deleteRowSchema.findField(name).fieldId())
                .collect(toImmutableList());
        for (int i =1;i<3;i++) {
            String deleteFile = "delete_file_" + UUID.randomUUID();
            Parquet.DeleteWriteBuilder writerBuilder = Parquet.writeDeletes(new ForwardingFileIo(fs).newOutputFile(new Path(metadataDir, deleteFile).toString()))
                    .forTable(icebergTable)
                    .rowSchema(deleteRowSchema)
                    .createWriterFunc(GenericParquetWriter::buildWriter)
                    .equalityFieldIds(equalityFieldIds)
                    .overwrite();
            EqualityDeleteWriter<Record> writer = writerBuilder.buildEqualityWriter();

            Record dataDelete = GenericRecord.create(deleteRowSchema);
            try (Closeable ignored = writer) {
                writer.write(dataDelete.copy(ImmutableMap.of("regionkey", 1L)));
            }
            icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
        }


        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE  regionkey > 2L");
        assertUpdate("DROP TABLE " + tableName);
    }
```


However, this detail is obvious only while debugging.

Probably a unit test for `IcebergPageSourceProvider` would make it clear and future proof.

Reply:
We need a unit test for `IcebergPageSourceProvider`. I will add this test for debug this problem.